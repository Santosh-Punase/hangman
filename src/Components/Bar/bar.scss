
@import 'src/assets/scss/variables';

.hangman__bar {
  background-color: $gray-dark;
  height: 1rem;
  border-top-right-radius: 0.4rem;
  border-bottom-right-radius: 0.4rem;
  position: relative;
  left: -2rem;

  &-container {
    align-self: flex-start;
    width: 100%;
  }

  &.bar-horizontal {
    width: 75%;
  }

  &.bar-vertical {
    width: 4rem;
    left: calc(50% - 2rem);
    transform: translateY(1rem) rotate(90deg);
  }
}
