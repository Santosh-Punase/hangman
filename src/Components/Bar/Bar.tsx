import React from 'react';

import './bar.scss';

const Bar: React.FC = () => {
  return (
    <div className="hangman__bar-container">
      <div className="hangman__bar bar-horizontal"></div>
      <div className="hangman__bar bar-vertical"></div>
    </div>
  );
}

export { Bar };
