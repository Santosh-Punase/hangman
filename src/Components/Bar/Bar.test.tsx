import React from 'react';
import { render, cleanup } from '@testing-library/react';

import { Bar } from './Bar';

afterEach(cleanup);

describe('Hangman bars', () => {
  test('should render bars', () => {
    const { asFragment } = render(<Bar />);
    
    expect(asFragment()).toMatchSnapshot();
  });
});
