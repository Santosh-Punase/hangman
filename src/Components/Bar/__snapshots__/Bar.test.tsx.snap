// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hangman bars should render bars 1`] = `
<DocumentFragment>
  <div
    class="hangman__bar-container"
  >
    <div
      class="hangman__bar bar-horizontal"
    />
    <div
      class="hangman__bar bar-vertical"
    />
  </div>
</DocumentFragment>
`;
